darts/timeseries.py

```diff
@@ -3005,6 +3037,598 @@ def map(
+                            ),
+                            function_name,
+                        )(**function_kwargs),
+                    ],
+                    axis=1,
+                )
+
+            elif isinstance(
+                transformation["function"], Callable
+            ):  # user provided function
+                new_columns.extend(
+                    [
+                        f"userFn{transformation['window'] if 'window' in transformation else idx}_{comp_name}"
+                        for comp_name in comps_to_transform
+                    ]
+                )
+                resulting_transformations = pd.concat(
+                    [
+                        resulting_transformations,
+                        ts_df[filter_df_columns]
+                        .rolling(**function_group_kwargs)
+                        .apply(fn, **function_kwargs),
+                    ],
+                    axis=1,
+                )
+
+            else:
+                raise_log(
+                    ValueError("The transformation function is not valid."), logger
+                )
+
+            # filter out non_transformed components
+            if keep_non_transformed:
+                # retrieve what needs to be kept but was not already kept (in case of multiple transformations)
+                to_add = (
+                    set(filter_df_non_transformed) - previously_added_non_transformed
+                )
+                resulting_transformations = pd.concat(
+                    [resulting_transformations, ts_df[list(to_add)]], axis=1
+                )
+                previously_added_non_transformed = (
+                    previously_added_non_transformed.union(to_add)
+                )  # update set
+                if not self.is_deterministic:
+                    new_columns.extend(
+                        {
+                            re.sub(f"{sample_prefix}.*$", "", comp_name)
+                            for comp_name in to_add
+                        }
+                    )
+                else:
+                    new_columns.extend(to_add)
+
+        # Treat NAs
+        if isinstance(treat_na, int) or isinstance(treat_na, float):
+            resulting_transformations.fillna(value=treat_na, inplace=True)
+            if not suppress_warnings:
+                raise_user_warning(
+                    forecasting_safe,
+                    f"NAs were replaced by provided value {treat_na}.",
+                    logger,
+                )
+        elif forecasting_safe:
+            resulting_transformations.dropna(inplace=True)
+            if not suppress_warnings:
+                raise_user_warning(
+                    forecasting_safe,
+                    "Enforcing forecasting safe. Resulting series is truncated by dropping NAs.",
+                    # TODO: ASSUMES NAs ARE AT THE BEGINNING OF THE SERIES, IS THIS A SAFE ASSUMPTION?
```

Comment:
Good point. We don't want to drop all NAs I think, as some might come later in the series. Could we only drop the first few time stamps affected (by looking at the window length, "center" and "closed"?